src/blueprint.zig

```diff
@@ -335,6 +337,116 @@ pub const Pattern = struct {
 	}
 };
 
+pub const Mask = struct {
+	entries: ListUnmanaged(Entry),
+
+	pub const separator = ',';
+	pub const inverse = '!';
+	pub const tag = '$';
+	pub const property = '@';
+
+	const Entry = struct {
+		inner: Inner,
+		isInverse: bool,
+
+		const Inner = union(enum) {
+			block: struct {typ: u16, data: ?u16},
```

Comment:
I would suggest to split this into two type `blockType` and `block`, then we don't need the optional.

Reply:
That optional implies whether the data match is mandatory or not how would I determine in suggested layout?

Reply:
I changed the implementation in away that outdates this suggestion, hence I will reject it for now.

Reply:
This suggestion still applies.

> That optional implies whether the data match is mandatory or not, how would I determine that in suggested layout?

Well you just have one separate tag for each case, instead of using an optional data field:
```zig
block: Block, // Matches type + data
blockType: u16, // Matches only the type
```